arrow/datatype_fixedwidth.go

```diff
@@ -532,19 +533,74 @@ type DecimalType interface {
 	DataType
 	GetPrecision() int32
 	GetScale() int32
+	BitWidth() int
 }
 
 func NewDecimalType(id Type, prec, scale int32) (DecimalType, error) {
 	switch id {
+	case DECIMAL32:
+		debug.Assert(prec <= int32(decimal.MaxPrecision[decimal.Decimal32]()), "invalid precision for decimal32")
```

Comment:
should this condition be an error? or is the `DecimalNType` informational

Reply:
The assumption being made is that if a user is explicitly calling `NewDecimalType` they know what they are doing, so we only perform the precision validation in debug mode for performance.